cpu/cc2538/include/cpu_conf.h

```diff
@@ -20,6 +20,9 @@
 #define CPU_CONF_H
 
 #include "cpu_conf_common.h"
+/* store compiler switches */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
```

Comment:
For this warning ignoring, I would prefer to see why its disabled than "store compiler switches".

Reply:
yes right, here and below its mostly to exclude vendor headers from being checked with pedantic on - because we don't want to change or fix them but rather use the originals and also update them from time to time